src/VisualStudio/Core/Def/Implementation/DesignerAttribute/DesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -200,49 +199,57 @@ private void RegisterDesignerAttribute(Document document, string designerAttribu
                 return;
             }
 
-            var documentId = document.Id;
-            _notificationService.RegisterNotification(() =>
+            var cps = await IsCpsProjectAsync(document.Project, cancellationToken).ConfigureAwait(false);
+            if (cps)
+            {
+                await _projectItemDesignerTypeUpdateService.SetProjectItemDesignerTypeAsync(document.FilePath, designerAttributeArgument).ConfigureAwait(false);
```

Comment:
How does this interface work? If you have the same file in multiple projects it'll just apply to all of them? Will we have some strange stuff if we're analyzing a linked/multitargeted file multiple times?

Reply:
ah. that is true. @davkean